includes/class-wc-product-variable.php

```diff
@@ -455,11 +455,15 @@ public static function sync( $product, $save = true ) {
 			$data_store->sync_stock_status( $product );
 			self::sync_attributes( $product ); // Legacy update of attributes.
 
-			do_action( 'woocommerce_variable_product_sync', $product->get_id(), $product->get_visible_children(), $save );
+			do_action( 'woocommerce_variable_product_sync_before_save', $product );
```

Comment:
Save is also an 'optional' thing. TBH I think a hook name such as `woocommerce_variable_product_sync_data` would be fine. We'd have used `woocommerce_variable_product_sync` if available but it wouldn't be backwards compatible.